### ISSUE TITLE
Idea: Export assemblies as STEP

### ISSUE THREAD

**Issue by Kobaltauge** [toptea/batch_process#2]
I found your script and I'm very impressed. I have a similar task, the only diffence is, that I don't need DXF files, I need STEP files.

How can the "export" command by modified, to export a STEP file?
self.doc.Export(str(path), enum)

**Comment by toptea** [comment]
Hi Massimo,

Hmm, it's been while! To be honest, this script is mainly tailored for me (and looking back, I think I've made the Document classes too complicated!). All you need to do is connect Inventor using the COM api, and change the file extension in the document's `SaveAs` function:

```python
import win32com.client


def application(silent=True, visible=True):
    mod = win32com.client.gencache.EnsureModule(
        '{D98A091D-3A0F-4C3E-B36E-61F62068D488}', 0, 1, 0)
    app = win32com.client.Dispatch('Inventor.Application')
    app = mod.Application(app)
    app.SilentOperation = silent
    app.Visible = visible
    return app


def load_document(path, app):
    document_type_enum = {
        12289: 'UnnownDocument',
        12290: 'PartDocument',
        12291: 'AssemblyDocument',
        12292: 'DrawingDocument',
        12293: 'PresentationDocument',
        12294: 'DesignElementDocument',
        12295: 'ForeignModelDocument',
        12296: 'SATFileDocument',
        12297: 'NoDocument',
    }
    app.Documents.Open(path)
    document_type = document_type_enum[app.ActiveDocumentType]
    doc = win32com.client.CastTo(app.ActiveDocument, document_type)
    return doc


if __name__ == '__main__':
    # change this!
    INPUT_PATH = 'C:/Users/GARY/Desktop/assembly.iam'
    OUTPUT_PATH = 'C:/Users/GARY/Desktop/assembly.stp'

    app = application()
    doc = load_document(INPUT_PATH, app)
    doc.SaveAs(OUTPUT_PATH, True)
```
Hope this helps!

**Comment by Kobaltauge** [comment]
Thank you. What a great and slim solution. Now I can deactivate the VBA script an go back to python.
Unfortunately in the meantime we changed the exchange format from step to jt. And 
doc.SaveAs and doc.SaveCopyAs does not create the JT file.

**Comment by toptea** [comment]
Can't help you there I'm afraid. We tend to convert all the drawings to pdf/dwf files, and send them for manufacturing.

If it's for design, we often find there's a few details missing when exporting from another program. (For example, when going from SolidWorks to Inventor, it just show the part as one solid body. Any additional modification will be hard to do on this piece). Unfortunately, it maybe better to re-model this part again using your choosing CAD program, and save it in it's native file.

If you still want to have a shot at this, try going into Inventor's Programming/API help and search for "Translator". There maybe some useful gem in there for you to work with. Good luck!

**Comment by Kobaltauge** [comment]
I know that remodeling is the best way. We have to migrate 200 assemblies including all parts from Inventor to NX. We found, that the berst format for this one time doing is JT. But if only STEP is possible, it wouldn't be a k.o. criteria.

As I wrote before, I have a VBA script running in Inventor that saves all assemblies and parts in STEP and JT. For a better automation of the complete conversion workflow it would be better to have all in the python scripts I wrote.

Thank you again for your help. And should I find a solution, I'll report it to you.

**Comment by Kobaltauge** [comment]
I found the solution. Better to say. There is no special solution.

After the doc.SaveAs(OUTPUT_PATH, True) for the STEP file an error occurse. Unfortunately I can't see what error ('Unspecified error\r\n' as error code).
But wiht this error the SaveAs for the JT File didn't execute. As I only tried to export the JT it worked properly only setting the OUTPUT_PATH as a ".jt" file name.

I don't know why the STEP export suddently do not work anymore. Probably a reboot of the system will help.

Thanks again!
